src/web/topic/components/TopicPane/TopicPane.tsx

```diff
@@ -11,76 +13,136 @@ import {
 } from "@/web/topic/components/TopicPane/TopicPane.styles";
 import { TopicViews } from "@/web/topic/components/TopicPane/TopicViews";
 import {
+  setIsDetailsOpen,
   setIsTopicPaneOpen,
+  setIsViewsOpen,
   setSelectedTab,
   usePaneStore,
+  useViewportWidth,
 } from "@/web/topic/components/TopicPane/paneStore";
 import { useSelectedGraphPart } from "@/web/view/currentViewStore/store";
 
-interface Props {
-  isLandscape: boolean;
+const IconAnchorMap = {
+  left: {
+    open: ChevronLeft,
+    close: AutoStories,
+  },
+  right: {
+    open: ChevronRight,
+    close: AutoStories,
+  },
+  bottom: {
+    open: KeyboardArrowDown,
+    close: AutoStories,
+  },
+};
+
+interface DrawerProps {
+  anchor: "left" | "right" | "bottom";
+  isOpen: boolean;
+  onToggleClick: () => void;
+  children: ReactNode;
 }
 
-export const TopicPane = ({ isLandscape }: Props) => {
-  const { isTopicPaneOpen, selectedTab } = usePaneStore();
+const TopicDrawer = ({ anchor, isOpen, onToggleClick, children }: DrawerProps) => {
+  const ToggleIcon = isOpen ? IconAnchorMap[anchor].open : IconAnchorMap[anchor].close;
+  const isLandscape = anchor === "right" || anchor === "left";
+  return (
+    <Box
+      sx={{
+        position: "absolute",
+        inset: "0",
+      }}
+    >
+      <PositionedDiv anchor={anchor}>
+        <StyledDrawer variant="permanent" open={isOpen} anchor={anchor} isLandscape={isLandscape}>
+          {children}
+        </StyledDrawer>
+        <TogglePaneButton color="primary" onClick={onToggleClick}>
+          <ToggleIcon />
+        </TogglePaneButton>
+      </PositionedDiv>
+    </Box>
+  );
+};
 
+export const TopicPane = () => {
+  const { isTopicPaneOpen, isViewsOpen, isDetailsOpen, selectedTab } = usePaneStore();
   const selectedGraphPart = useSelectedGraphPart();
+  const isLandscape = useMediaQuery("(orientation: landscape)");
+  const width = useViewportWidth();
```

Comment:
I programed it because it was an easy implementation, I didn't feel the need to search through the imported libraries for a possible solution. If your preference is to use MUI ill change it over.